operator/pkg/reconciliation/reconcile_datacenter.go

```diff
@@ -25,6 +26,14 @@ func (rc *ReconciliationContext) ProcessDeletion() result.ReconcileResult {
 		return result.Error(err)
 	}
 
+	// Clean up annotation litter on the user Secrets
+	err := rc.SecretWatches.RemoveWatcher(types.NamespacedName{
+		Name: rc.Datacenter.GetName(), Namespace: rc.Datacenter.GetNamespace(),})
+
+	if err != nil {
+		rc.ReqLogger.Error(err, "Failed to remove dynamic secret watches for CassandraDatacenter")
```

Comment:
Looking for feedback. Should we return an error here or just log it? It seems wrong to hold up a delete by returning an error, but maybe that is the right thing to do.

Reply:
Yeah, let's just log it.  Otherwise, how would someone remove a dc that has some sort of error that prevents removing the watch, right?

Reply:
I'm not so sure, but I'm fine with trying this as you've coded it up